dataflow/run_template/main_test.py

```diff
@@ -44,29 +46,72 @@ def app():
     return flask.Flask(__name__)
 
 
-def unique_job_name(label):
-    return datetime.now().strftime('{}-%Y%m%d-%H%M%S-{}'.format(
-        label, uuid.uuid4().hex))
+# fixture used to create user-friendly names with UUIDs for Dataflow jobs
+# indirect parametrization inspired by this post https://stackoverflow.com/questions/42228895/how-to-parametrize-a-pytest-fixture
+@pytest.fixture(scope="function")
+def dataflow_job_name(request):
+    label = request.param
+    job_name = datetime.now().strftime('{}-%Y%m%d-%H%M%S-{}'.format(
+        label, uuid.uuid4().hex[:5]))
+
+    yield job_name
+
+    # cancel the Dataflow job after running the test
+    dataflow_jobs_cancel(job_name)
+
+
+# Takes in a Dataflow job name and returns its job ID
+def get_job_id_from_name(job_name):
+    # list the 100 most recent Dataflow jobs
+    jobs_request = dataflow.projects().jobs().list(
+        projectId=PROJECT,
+        pageSize=50  # only return the 50 most recent results - our job is likely to be in here. If the job is not found, first try increasing this number
```

Comment:
Maybe also add `filter` to only get active jobs?

Reply:
good idea